test/unit/deploy.js

```diff
@@ -160,6 +160,57 @@ exports['Tessel.prototype.deployScript'] = {
       test.done();
     }.bind(this));
   },
+
+  processCompletionOrder: function(test) {
```

Comment:
No, this is a test that will fail if we ever have a regression back to the issue that this PR fixes.

The issue was that the project contents were being written to `/tmp/remote-script` before the process that created that folder completes. This test ensures that each step of the deployment process (including folder creation) completes before the previous starts.